crates/biome_js_formatter/src/utils/binary_like_expression.rs

```diff
@@ -530,6 +530,7 @@ impl AnyJsBinaryLikeExpression {
                             | JsSyntaxKind::JS_THROW_STATEMENT
                             | JsSyntaxKind::JS_CALL_EXPRESSION
                             | JsSyntaxKind::JS_IMPORT_CALL_EXPRESSION
+                            | JsSyntaxKind::JS_CALL_ARGUMENT_LIST
```

Comment:
When a binary expression is part of the call _arguments_, its grand parent is `JsCallArgumentList`, not just `JsCallExpression`. These intermediary nodes make it kind of hard to compare directly to prettier, but come up consistently (other places also have to check like `JsParameterList` instead of `JsFunctionDeclaration`, for example).

Adding this here means that binaries are formatted consistently whether they're the receiver or an argument to a call expression:

```typescript
foo(
  someCondition === true ||
  anotherCondition === false
);
```
should actually indent the right hand side, but wasn't because the grandparent is `JsCallArgumentList`, but this was only checking for `JsCallExpression` directly. Now it matches Prettier as:
```typescript
foo(
  someCondition === true ||
    anotherCondition === false
);
```